Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -221,149 +262,258 @@ public List<InteractableEvent> InteractableEvents
             }
         }
 
+        private List<InteractableThemeBase> activeThemes = new List<InteractableThemeBase>();
+
         /// <summary>
         /// The list of running theme instances to receive state changes
         /// When the dimension index changes, the list of themes that are updated changes to those assigned to that dimension.
         /// </summary>
-        private List<InteractableThemeBase> activeThemes = new List<InteractableThemeBase>();
+        public IReadOnlyList<InteractableThemeBase> ActiveThemes => activeThemes.AsReadOnly();
 
-        // directly manipulate a theme value, skip blending
-        protected bool forceUpdate = false;
+        /// <summary>
+        /// How many times this interactable was clicked
+        /// </summary>
+        /// <remarks>
+        /// Useful for checking when a click event occurs.
+        /// </remarks>
+        public int ClickCount { get; private set; }
 
         #region States
 
+        // Field just used for serialization to save if the Interactable should start enabled or disabled
+        [FormerlySerializedAs("Enabled")]
+        [SerializeField]
+        private bool enabledOnStart = true;
+
         /// <summary>
-        /// Has focus
+        /// Defines whether the Interactable is enabled or not internally
+        /// This is different than the Enabled property at the GameObject/Component level
+        /// When false, Interactable will continue to run in Unity but not respond to Input.
         /// </summary>
-        public bool HasFocus { get; private set; }
+        /// <remarks>
+        /// Property is useful for disabling UX, such as greying out a button, until a user completes some pre-mandatory step such as fill out their name, etc
+        /// </remarks>
+        public virtual bool IsEnabled
+        {
+            // Note the inverse setting since targeting "Disable" state but property is concerning "Enabled"
+            get { return !(GetStateValue(InteractableStates.InteractableStateEnum.Disabled) > 0); }
+            set
+            {
+                if (IsEnabled != value)
+                {
+                    // If we are disabling input, we should reset our base input tracking states since we will not be responding to input while disabled
+                    if (!value)
+                    {
+                        ResetBaseStates();
```

Comment:
>esetBaseStates(); [](start = 25, length = 17)

global listeners should also be unregistered if the interactable is disabled

Reply:
No, that only occurs in OnDisable. When Interactable.IsEnabled = false but the GameObject is still enabled, we still see the item and can still collide our cursors on it etc.

We still accept global input but we just ignore processing it via CanInteract() which is largely powered by IsEnabled.

I like explicitly controlling global register/unregister in the tight loop of OnEnable/OnDisable. When we start changing that via other code entrance paths then things get more complicated and prone to issues

GameObject.enable => Means will this thing run at all. More all or nothing
Interactable.enable => means will I ACCEPT input. I'll still update my themes and listen etc, but will I respond